play-interop-test-scala/src/test/scala/play/api/test/NewWithServer.scala

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package play.api.test
+
+import org.specs2.execute.{ AsResult, Result }
+import org.specs2.mutable.Around
+import org.specs2.specification.Scope
+import play.api.Application
+import play.api.inject.guice.GuiceApplicationBuilder
+
+// RICH: replacement for play-specs2's WithServer class; adding serverEndpoints
+// RICH: the constructor has been refactored; when merging into Play there will
+// need to be more consideration for backwards compatibility (should be doable)
+abstract class NewWithServer(
+  val app: Application = GuiceApplicationBuilder().build(),
+  testServerFactory: TestServerFactory = new DefaultTestServerFactory) extends Around with Scope {
+
+  implicit def implicitMaterializer = app.materializer
+  implicit def implicitApp = app
+
+  /**
+   * The list of endpoints of the currently running test server.
+   */
+  protected def serverEndpoints: ServerEndpoints = synchronized {
```

Comment:
The main change: support for multiple server endpoints, e.g. both HTTP and HTTPS.